aisee/vision_classifier.py

```diff
@@ -601,3 +602,150 @@ def predict_loop(
+            +------------------------+---------+---------------+--------+
+            |"sample/vehicles3.png"  |    1    |       0       |    1   |
+            +------------------------+---------+---------------+--------+
+            |"sample/vehicles4.png"  |    1    |       0       |    0   |
+            +------------------------+---------+---------------+--------+
+
+        - If it is a string, it must be:
+            - A path to an image.
+
+            Example
+
+            .. code-block:: console
+
+                /data/cat.png
+
+            - A directory where each folder is a class, and inside that class are the
+            images.
+
+            Example:
+
+            .. code-block:: console
+
+                ├── animals
+                ├── cat
+                │  ├── cat1.jpg
+                │  └── cat2.jpg
+                └── dog
+                    ├── dog1.jpg
+                    └── dog2.jpg
+
+        metrics : List[Callable]
+            Each element of the list is a function that at least has the parameters
+            `y_pred` and `y_true`, and each parameter accepts pure predictions as 1D array-like.
+
+            For example, you can import `accuracy_score` from `sklearn.metrics`.
+        kwargs : Dict[str, Dict[str, Any]]
+            Each key of the dictionary represents the name of one of the functions
+            indicated in the `metrics` parameter. The value is a dictionary with the
+            arguments of thath function.
+
+            For example, if your `metrics` parameter has the function `f1_score` from
+            `sklearn.metrics` and you want to use the parameter `average` with `micro` value,
+            your kwargs will be:
+
+                .. code-block:: python
+
+                kwargs = {"f1_score": {"average": "micro"}}
+
+        num_workers : int, default=2
+            Subprocesses to use for data loading.
+        data_transform : torchvision.transforms.Compose, default=None
+        batch_size : int, default=8
+
+        Returns
+        -------
+        evaluation_results : Dict[str, Any]
+            The resulting dictionary has a key for each function in the `metrics` parameter.
+            The values are the results of each function.
+        """
+        predictions = self.predict(data, num_workers, data_transform, batch_size)
+        predictions = pd.DataFrame(predictions, columns=["prediction", "real_label"])
+
+        if self.task == "single_label":
+            y_pred = [value.item() for value in predictions["prediction"].to_numpy()]
+            y_true = [value.item() for value in predictions["real_label"].to_numpy()]
+        elif self.task == "multi_label":
+            y_pred = np.concatenate(predictions["prediction"].to_numpy())
+            y_true = np.concatenate(predictions["real_label"].to_numpy())
+        else:
+            raise NotImplementedError(
+                f"""Evaluate method is not implemented for task {self.task}""",
+            )
```

Comment:
This was not covered by the tests because it is impossible to happen. When we create the class `VisionClassifier`, we already check the task so this else is redundant.

```suggestion
```

Reply:
This is a check in case we implement a new task in the future and we don't update the evaluate method

Reply:
We have two options then. Either we make sure that when we implement the new task, we make it work for the evaluate or we test this line of code, which would be hard right now

Reply:
Hmmm, we can remove this check. However there is a test that should fail is something is wrong with the evaluate method

Reply:
The only thing that could go wrong is that the user uses a non-standard metric. In that case, the error may be confusing. We could check that the provided metrics are ok maybe.

What else do you have in mind that could go wrong?

Reply:
Do you  mean checking that the function metrics accept y_true and y_pred?

Reply:
Yes

Reply:
Added the parameters check and corresponding test. Also removed the "Task not implemented" check